### PR TITLE
Seed permissions

### PR DESCRIPTION
**Pull request by JMurphyWeb**
# ready

Adds seed data

#10

### CODE REVIEW

File: lib/ev2/seed/accounts.ex [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  @doc """
+  This function gets called by the outer world. Seeds all default accounts data.
+  """
+  def seed_all() do
+    add_permissions()
+    add_roles()
+    add_role_permissions()
+  end
+
+  def add_permissions() do
```
Comment (jackcarlisle) [inline lib/ev2/seed/accounts.ex]: Could refactor `add_permissions()` and `add_roles()` into one general function

File: lib/ev2/accounts/role_api.ex [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+
+  @doc """
+  get a role by id
+  """
+  def get_role!(id) do
+    Repo.get!(Role, id)
+  end
+
+  def get_by_name(name) do
```
Comment (jackcarlisle) [inline lib/ev2/accounts/role_api.ex]: We could potentially move these functions into a `seed_helpers.ex` file if they are specific to seeding

File: lib/ev2/seed/accounts.ex (hunk 1 above)
Comment (jackcarlisle) [reply lib/ev2/seed/accounts.ex]: https://github.com/karmaradio/ev2/pull/12/files#diff-5882caa45d668f9358926d58cea2c6a8R20
+
https://github.com/karmaradio/ev2/pull/12/files#diff-5882caa45d668f9358926d58cea2c6a8R30

File: lib/ev2/seed/accounts.ex [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  insert_list(
+    [%{name: "can_view_offer"}, ... ],
+    Permission
+  )
+  """
+  def insert_list(data, schema) do
+    Ev2.Repo.insert_all(schema, data)
+  end
+
+  defp get_difference(seed_data, db_data) do
```
Comment (jackcarlisle) [inline lib/ev2/seed/accounts.ex]: could maybe add a comment what this function is getting the difference between and why

File: lib/ev2/seed/accounts_data.ex [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    When the need for editing data arrives, we need to re-write the
+    seed helper functions in `Ev2.Seed.Accounts`
+  """
+
+  @doc """
+  Seed permissions.
+  This structure should probably change to an array of strings but
+  this was the quickest way for now.
+  """
+  def permissions() do
```
Comment (jackcarlisle) [inline lib/ev2/seed/accounts_data.ex]: make permissions uppercase

**Comment by jackcarlisle** [comment]
This might be a good time to try and integrate seeding in our review apps if we can figure it out

File: lib/ev2/seed/accounts_data.ex (hunk 4 above)
Comment (JMurphyWeb) [reply lib/ev2/seed/accounts_data.ex]: Done

**Pull request merged**